docs/source/workflow/generate-types.mdx

```diff
@@ -0,0 +1,205 @@
+
+If you haven't already, begin by installing the `graphql-tag` library:
+
+```bash
+npm install graphql-tag
+```
+
+The `gql` template literal tag from `graphql-tag` can parse a Schema Definition Language (SDL) string into an abstract syntax tree (AST). Apollo Server accepts a string or an AST as a valid option for defining a schema.  
+
+If we pass GraphQL Code Generator a TypeScript file path, it will automatically generate types for the `gql` tag within that file. So, we can wrap our schema string in a `gql` tag to let GraphQL Code Generator know it should generate types for that tag. 
+
+Import the `gql` tag into the file where you define your schema, then wrap your schema string like so:
+
+```ts title="src/schema.ts"
+import gql from 'graphql-tag';
+
+export default gql`
+  type Query {
+    books: [Book]
+  }
+
+  type Book {
+    title: String
+    author: String
+  }
+
+  type AddBookMutationResponse {
+    code: String!
+    success: Boolean!
+    message: String!
+    book: Book
+  }
+
+  type Mutation {
+    addBook(title: String, author: String): AddBookMutationResponse
+  }
+`;
+```
+
+Restart your server to ensure everything still works as expected. Next, we'll install the packages we need to generate types automatically based on our schema.
+
+### Installing and configuring dependencies
+
+Run the following command to install the `@graphql-codegen/cli`, `@graphql-codegen/typescript`, and `@graphql-codegen/typescript-resolvers` packages into your project's dev dependencies:
+
+```bash
+npm install -D @graphql-codegen/cli @graphql-codegen/typescript @graphql-codegen/typescript-resolvers
+```
+
+> For more information on each package above, check out the [GraphQL Code Generator](https://www.the-guild.dev/graphql/codegen/plugins) docs.
+
+Next, we'll set up a configuration file to tell GraphQL Code Generator where and how to generate types. You can do this by manually creating a `codegen.yml` file or by using the following command, which walks you through the process:
+
+```bash
+npx graphql-code-generator init
+```
+
+Below is an example of a `codegen.yml` file:
+
+```yaml
+# This configuration file tells GraphQL Code Generator how
+# to generate types based on our schema.
+schema: "./src/schema.ts"
+generates:
+  # Specify where our generated types should live.
+  ./src/__generated__/resolvers-types.ts:
+    plugins:
+      - "typescript"
+      - "typescript-resolvers"
+    config:
+      useIndexSignature: true
+      # More on this below!
+      contextType: ../index#MyContext
+```
+
+> [See the docs](https://www.the-guild.dev/graphql/codegen/plugins/typescript-resolvers#usage-example) for more information on the above configuration options.
+
+Finally, we recommend adding helpful scripts to your `package.json` file to ensure your TS types are regularly generated:
+
+```json
+{
+// ...
+  "scripts": {
+    "generate": "graphql-codegen --config codegen.yml",
```

Comment:
we recommend also adding a `"generate:watch": "graphql-codegen --watch --config codegen.yml"` command for dev purposes so typed get updated when modifying the schema

Reply:
I think this would be a good addition, along with a `compile:watch` and a top-level `watch` command which runs both in parallel. Explaining them following the sentence below would be good.

Reply:
We do include the `watch` command in the [accompanying docs example](https://github.com/apollographql/docs-examples/blob/e3c133e50cd1d9ce6fb54630ca9ea9395b9bfa4d/apollo-server/v4/generated-types/package.json#L12) which generates types using `graphql-codegen -w` , compiles TS, and updates the server using `Nodemon`.

Reply:
Oops, GH didn't load your comment @trevor-scheer 😂  Okay, I can add those additional scripts if we feel like they'll be helpful for folks!

Reply:
@rkoron007 I'm pretty happy keeping this doc simple and having additional complexity like that exist in the example, so I'll leave it up to you!

Reply:
I personally like keeping the article simpler and adding the `watch` command in the example, because having a `watch` script isn't _required_. I can add a note to this article pointing to the `watch` command in the example though. That feels like the best of both worlds to me ✨